aioredlock/algorithm.py

```diff
@@ -171,34 +174,45 @@ async def extend(self, lock, lock_timeout=None):
                 await self.unlock(lock)
             raise
 
-    async def unlock(self, lock):
+    async def unlock(self, resource_or_lock, lock_identifier=None):
```

Comment:
I think that we should still just be passing in the lock here, now that you can get a recreated lock.  This was the whole point of adding `get_lock` to leave `unlock` unmodified.